compiler/plugins/input/Torch/InputConversion/ConvertTMTensorToLinalgExt.cpp

```diff
@@ -147,18 +155,22 @@ struct AttentionOpConversion
         loc, targetType, rewriter.getFloatAttr(targetType, dk));
 
     // Add batches to standard attention indexing maps.
-    SmallVector<AffineMap> indexingMaps = getStandardAttentionIndexingMaps(ctx);
+    SmallVector<AffineMap> indexingMaps =
+        getStandardAttentionIndexingMaps(ctx, optionalMask.has_value());
+
     int64_t numBatches = op.getQueryType().getRank() - 2;
     for (AffineMap &map : indexingMaps) {
       map = map.shiftDims(numBatches);
-      for (int batch : llvm::seq<int>(numBatches)) {
-        map = map.insertResult(rewriter.getAffineDimExpr(batch), batch);
+      if (map.getNumResults() > 0) {
+        for (int batch : llvm::seq<int>(numBatches)) {
+          map = map.insertResult(rewriter.getAffineDimExpr(batch), batch);
+        }
```

Comment:
style nit: we prefer early-exit which saves the level of nesting.

```suggestion
      if (map.getNumResults() == 0) continue;
      for (int batch : llvm::seq<int>(numBatches)) {
        map = map.insertResult(rewriter.getAffineDimExpr(batch), batch);
```

Reply:
Makes sense